scene/gui/graph_frame.cpp

```diff
@@ -0,0 +1,357 @@
+#include "scene/theme/theme_db.h"
+
+void GraphFrame::gui_input(const Ref<InputEvent> &p_ev) {
+	ERR_FAIL_COND(p_ev.is_null());
+
+	Ref<InputEventMouseButton> mb = p_ev;
+	if (mb.is_valid()) {
+		ERR_FAIL_NULL_MSG(get_parent_control(), "GraphFrame must be the child of a GraphEdit node.");
+
+		if (mb->is_pressed() && mb->get_button_index() == MouseButton::LEFT) {
+			Vector2 mpos = mb->get_position();
+
+			Ref<Texture2D> resizer = theme_cache.resizer;
+
+			if (resizable && mpos.x > get_size().x - resizer->get_width() && mpos.y > get_size().y - resizer->get_height()) {
+				resizing = true;
+				resizing_from = mpos;
+				resizing_from_size = get_size();
+				accept_event();
+				return;
+			}
+
+			emit_signal(SNAME("raise_request"));
+		}
+
+		if (!mb->is_pressed() && mb->get_button_index() == MouseButton::LEFT) {
+			if (resizing) {
+				resizing = false;
+				emit_signal(SNAME("resize_end"), get_size());
+				return;
+			}
+		}
+	}
+
+	Ref<InputEventMouseMotion> mm = p_ev;
+
+	// Only resize if the frame is not auto-resizing based on linked nodes.
+	if (resizing && !autoshrink_enabled && mm.is_valid()) {
+		Vector2 mpos = mm->get_position();
+
+		Vector2 diff = mpos - resizing_from;
+
+		emit_signal(SNAME("resize_request"), resizing_from_size + diff);
+	}
+}
+
+Control::CursorShape GraphFrame::get_cursor_shape(const Point2 &p_pos) const {
+	if (resizable && !autoshrink_enabled) {
+		if (resizing || (p_pos.x > get_size().x - theme_cache.resizer->get_width() && p_pos.y > get_size().y - theme_cache.resizer->get_height())) {
+			return CURSOR_FDIAGSIZE;
+		}
+	}
+
+	return Control::get_cursor_shape(p_pos);
+}
+
+void GraphFrame::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_DRAW: {
+			// Used for layout calculations.
+			Ref<StyleBox> sb_panel = theme_cache.panel;
+			Ref<StyleBox> sb_titlebar = theme_cache.titlebar;
+
+			// Used for drawing.
+			Ref<StyleBox> sb_to_draw_panel = selected ? theme_cache.panel_selected : sb_panel;
+			Ref<StyleBox> sb_to_draw_titlebar = selected ? theme_cache.titlebar_selected : sb_titlebar;
+			Ref<StyleBoxFlat> sb_panel_flat = sb_to_draw_panel;
+			Ref<StyleBoxTexture> sb_panel_texture = sb_to_draw_panel;
+
+			Rect2 titlebar_rect(Point2(), titlebar_hbox->get_size() + sb_titlebar->get_minimum_size());
+			Size2 body_size = get_size();
+			body_size.y -= titlebar_rect.size.height;
+			Rect2 body_rect(Point2(0, titlebar_rect.size.height), body_size);
+
+			// Draw body stylebox.
+			if (tint_color_enabled) {
+				if (sb_panel_flat.is_valid()) {
+					Color original_border_color = sb_panel_flat->get_border_color();
+					sb_panel_flat = sb_panel_flat->duplicate();
+					sb_panel_flat->set_bg_color(tint_color);
+					sb_panel_flat->set_border_color(selected ? original_border_color : tint_color.lightened(0.3));
+					draw_style_box(sb_panel_flat, body_rect);
+				} else if (sb_panel_texture.is_valid()) {
+					sb_panel_texture = sb_panel_flat->duplicate();
```

Comment:
This code might crash, as highlighted by its failure to compile on my Linux CI: https://github.com/Zylann/godot_voxel/actions/runs/8560927646/job/23461023064

Reply:
Reading the code, I think the two identifiers might be swapped bg mistake?